### PR TITLE
M2-4478: M2-4486: DeleteApplet and VersionHistory settings tests

### PR DESCRIPTION
**Pull request by wildkiomi**

### CODE REVIEW

File: src/shared/features/AppletSettings/DeleteAppletSetting/DeleteAppletSetting.test.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    mockAxios.delete.mockResolvedValueOnce(null);
+    jest.spyOn(encryptionFunctions, 'getAppletEncryptionInfo').mockImplementation(() => ({
+      getPublicKey: getPublicKeyMock,
+    }));
+    renderWithProviders(<DeleteAppletSetting />, { preloadedState, route, routePath });
+
+    expect(screen.getByTestId('applet-settings-delete-applet-delete-button')).toBeVisible();
+    expect(screen.getByTestId('applet-settings-delete-applet-delete-password-popup')).toBeVisible();
+
+    fireEvent.change(screen.getByLabelText(/Password/), {
```
Comment (alex-i-e) [inline src/shared/features/AppletSettings/DeleteAppletSetting/DeleteAppletSetting.test.tsx]: why not used `userEvent.type(screen.getByLabelText(/Password/), mockedPassword)`?

[@testing-library/user-event](https://github.com/testing-library/user-event) is a package that's built on top of fireEvent, but it provides several methods that resemble the user interactions more closely. In the example above, fireEvent.change will simply trigger a single change event on the input. However the type call, will trigger keyDown, keyPress, and keyUp events for each character as well. It's much closer to the user's actual interactions. This has the benefit of working well with libraries that you may use which don't actually listen for the change event.

File: src/shared/features/AppletSettings/VersionHistorySetting/VersionHistorySetting.test.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      renderWithProviders(<VersionHistorySetting />, { preloadedState, route, routePath });
+
+      expect(screen.getByTestId('spinner')).toBeVisible();
+      await waitFor(() => expect(screen.getByTestId(`${dataTestid}-version`)).toBeVisible());
+      expect(screen.getByText('current (2.0.0)')).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-applet-changes`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-activities-changes`)).toBeVisible();
+
+      const input = screen.getByTestId(`${dataTestid}-version`).querySelector('input');
+      input && fireEvent.change(input, { target: { value: '1.1.0' } });
```
Comment (alex-i-e) [inline src/shared/features/AppletSettings/VersionHistorySetting/VersionHistorySetting.test.tsx:107]: please use `userEvent` library for it, it's more closer to user interaction

File: src/shared/features/AppletSettings/DeleteAppletSetting/DeleteAppletSetting.test.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      target: { value: mockedPassword },
+    });
+
+    fireEvent.click(screen.getByText('Delete'));
+    await waitFor(() => fireEvent.click(screen.getByText('Ok')));
```
Comment (alex-i-e) [inline src/shared/features/AppletSettings/DeleteAppletSetting/DeleteAppletSetting.test.tsx]: no need `await waitFor(...) wrapper` , use `await screen.find*` instead

File: src/shared/features/AppletSettings/VersionHistorySetting/VersionHistorySetting.test.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      route                                                                     | routePath                         | description
+      ${`/dashboard/${mockedAppletId}/settings/${SettingParam.VersionHistory}`} | ${page.appletSettingsItem}        | ${'for dashboard'}
+      ${`/builder/${mockedAppletId}/settings/${SettingParam.VersionHistory}`}   | ${page.builderAppletSettingsItem} | ${'for builder'}
+    `('$description', async ({ route, routePath }) => {
+      mockAxios.get.mockResolvedValueOnce(versionsMock);
+      mockAxios.get.mockResolvedValueOnce(changesMock);
+      renderWithProviders(<VersionHistorySetting />, { preloadedState, route, routePath });
+
+      expect(screen.getByTestId('spinner')).toBeVisible();
+      await waitFor(() => expect(screen.getByTestId(`${dataTestid}-version`)).toBeVisible());
```
Comment (alex-i-e) [inline src/shared/features/AppletSettings/VersionHistorySetting/VersionHistorySetting.test.tsx:101]: here and in row 109 the same:
no need `await waitFor(...) wrapper` , use `await screen.find*` instead

**Review by IhorNesterenko99 — APPROVED** [review]

File: src/shared/features/AppletSettings/DeleteAppletSetting/DeleteAppletSetting.test.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  test('should render and submit', async () => {
+    mockAxios.delete.mockResolvedValueOnce(null);
+    jest.spyOn(encryptionFunctions, 'getAppletEncryptionInfo').mockImplementation(() => ({
+      getPublicKey: getPublicKeyMock,
+    }));
+    renderWithProviders(<DeleteAppletSetting />, { preloadedState, route, routePath });
+
+    expect(screen.getByTestId('applet-settings-delete-applet-delete-button')).toBeVisible();
+    expect(screen.getByTestId('applet-settings-delete-applet-delete-password-popup')).toBeVisible();
+
```
Comment (IhorNesterenko99) [inline src/shared/features/AppletSettings/DeleteAppletSetting/DeleteAppletSetting.test.tsx:72]: Could you pls add test for incorrect password error?
image

Comment (wildkiomi) [reply src/shared/features/AppletSettings/DeleteAppletSetting/DeleteAppletSetting.test.tsx:72]: such tests should be for DeletePopup component, not DeleteAppletSetting

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**